.ci/build_docs.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -xueo pipefail
+
+# Check that documentation was generated successfully
+if [[ "$GHC_VERSION" = "8.6.5" ]]; then
+  haddock_pkgs="clash-lib clash-cosim"
+else
+  haddock_pkgs="clash-prelude clash-lib clash-cosim"
+fi
+
+mkdir -p hadocs
+
+for pkg in ${haddock_pkgs}; do
+  set -e
+
+  # Cache dependencies
+  cabal v2-build ${pkg} -O0 --enable-documentation --only-dependencies
+
+  # HaskellPrelude yields warnings we cannot fix
+  cabal v2-haddock ${pkg} -O0 --enable-documentation \
+    |& grep -v "Clash.HaskellPrelude" \
+    |  tee haddock_log
+
+  set +e
+
+  # TODO: We're not yet ready for the check below :-)
```

Comment:
Are we still not ready for this check? If it relates to some issue can you provide the link in the TODO

Reply:
Oh no, it would require documentation for each and every construct in `clash-prelude`. I've done this for `clash-protocols` from the start and then it's manageable. But I don't look forward to going through our current codebase and get a 100% coverage..

Reply:
Understandable, can you update the comment to make that clearer?